br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -54,16 +65,57 @@ func openStoreReaderAndSeek(
 
 // newByteReader wraps readNBytes functionality to storageReader. It will not
 // close storageReader when meet error.
-func newByteReader(ctx context.Context, storageReader storage.ReadSeekCloser, bufSize int) (*byteReader, error) {
+func newByteReader(ctx context.Context, storageReader storage.ReadSeekCloser, bufSize int, st storage.ExternalStorage, name string, defaultUseConcurrency bool) (*byteReader, error) {
+	conReader, err := NewSingeFileReader(ctx, st, name, 8, ConcurrentReaderBufferSize)
```

Comment:
the new functions in this PR are no need to be public? Because `byteReader` is not public so caller still can't access them